### PR TITLE
Fix connection issue when providing invalid DNS and valid IP

### PR DESCRIPTION
**Pull request by shrikant1407**
* Fix connection issue when providing invalid DNS and valid IP
* Enable SNI by default
* Formatting the code

### CODE REVIEW

**Review by DukeDavis12 — DISMISSED** [review]

File: lib/fdoprotctx.c [hunk 1]
```diff
@@ -165,14 +163,15 @@ static bool fdo_prot_ctx_connect(fdo_prot_ctx_t *prot_ctx)
 	case FDO_STATE_TO1_SND_PROVE_TO_FDO: /* type 32 */
 		ATTRIBUTE_FALLTHROUGH;
 	case FDO_STATE_TO1_RCV_FDO_REDIRECT: /* type 33 */
-		// try DNS's resolved IP first, if it fails, try given IP address
+		// try DNS's resolved IP first, if it fails, try given IP
+		// address
 		ret = connect_to_rendezvous(
-		    prot_ctx->resolved_ip, prot_ctx->host_dns, prot_ctx->host_port,
-			&prot_ctx->sock_hdl, prot_ctx->tls);
+		    prot_ctx->resolved_ip, prot_ctx->host_dns,
```
Comment (tajnisha) [inline lib/fdoprotctx.c:169]: What will be this host_dns value in case we use only IP address (not domain name) ?

Comment (shrikant1407) [reply lib/fdoprotctx.c:169]: It will take IP only

File: cmake/cli_input.cmake [hunk 2]
```diff
@@ -16,7 +16,7 @@ set (TARGET_OS linux)
 set (HTTPPROXY true)
 set (PROXY_DISCOVERY false)
 set (SELF_SIGNED_CERTS true)
-set (SNI false)
+set (SNI true)
```
[the remark this answers is not in the document]
Comment (shrikant1407) [reply cmake/cli_input.cmake:19]: yes

File: lib/fdoprotctx.c [hunk 3]
```diff
@@ -165,14 +163,15 @@ static bool fdo_prot_ctx_connect(fdo_prot_ctx_t *prot_ctx)
...
+		    prot_ctx->host_port, &prot_ctx->sock_hdl, prot_ctx->tls);
 		if (!ret) {
 			ret = connect_to_rendezvous(
-				prot_ctx->host_ip, prot_ctx->host_dns, prot_ctx->host_port, 
-				&prot_ctx->sock_hdl, prot_ctx->tls);
+			    prot_ctx->host_ip, NULL, prot_ctx->host_port,
```
Comment (tajnisha) [inline lib/fdoprotctx.c:173]: if resolved_ip is not working, the second call seem to try with host_ip in current implementation. Now we are adding one another fallback combination to the same call. Will it be one another fallback call ?

File: lib/fdoprotctx.c [hunk 4]
```diff
@@ -139,11 +139,10 @@ static bool fdo_prot_ctx_connect(fdo_prot_ctx_t *prot_ctx)
...
 	case FDO_STATE_DI_DONE: /* type 13 */
 		ret = connect_to_manufacturer(
-			      prot_ctx->resolved_ip ? prot_ctx->resolved_ip : prot_ctx->host_ip,
-				  prot_ctx->host_dns,
-			      prot_ctx->host_port,
-			      &prot_ctx->sock_hdl,
-			      prot_ctx->tls);
+		    prot_ctx->resolved_ip ? prot_ctx->resolved_ip
+					  : prot_ctx->host_ip,
+		    prot_ctx->host_dns, prot_ctx->host_port,
```
Comment (tajnisha) [inline lib/fdoprotctx.c:144]: Should prot_ctx->host_dns be NULL in case of host_ip here as well ?

**Review by tajnisha — APPROVED** [review]

File: lib/fdoprotctx.c (hunk 1 above)
Comment (tajnisha) [reply lib/fdoprotctx.c:169]: As per discussed, hosted RV requires this SNI to be enabled and supported from device as the rv cert contain DNS. So we can have SNI enabled by default.

**Review by KiranSukhavasi — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

File: lib/fdoprotctx.c [hunk 5]
```diff
@@ -170,9 +170,8 @@ static bool fdo_prot_ctx_connect(fdo_prot_ctx_t *prot_ctx)
 		    prot_ctx->host_port, &prot_ctx->sock_hdl, prot_ctx->tls);
 		if (!ret) {
 			ret = connect_to_rendezvous(
-			    prot_ctx->host_ip, prot_ctx->host_dns,
-			    prot_ctx->host_port, &prot_ctx->sock_hdl,
-			    prot_ctx->tls);
+			    prot_ctx->host_ip, NULL, prot_ctx->host_port,
```
Comment (v-vasavi) [inline lib/fdoprotctx.c:173]: what is the impact of removing prot_ctx->host_dns? and adding null?

File: lib/fdoprotctx.c [hunk 6]
```diff
@@ -217,9 +216,8 @@ static bool fdo_prot_ctx_connect(fdo_prot_ctx_t *prot_ctx)
 				       &prot_ctx->sock_hdl, prot_ctx->tls);
 		if (!ret) {
 			ret = connect_to_owner(
-			    prot_ctx->host_ip, prot_ctx->host_dns,
-			    prot_ctx->host_port, &prot_ctx->sock_hdl,
-			    prot_ctx->tls);
+			    prot_ctx->host_ip, NULL, prot_ctx->host_port,
```
Comment (v-vasavi) [inline lib/fdoprotctx.c:219]: same as above

File: lib/fdoprotctx.c (hunk 5 above)
Comment (shrikant1407) [reply lib/fdoprotctx.c:173]: when providing invalid DNS (or NULL DNS) and valid IP it should connect with IP only so we make invalid DNS as NULL and go with IP

**Pull request merged**